src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -121,24 +127,46 @@ private String getHIUId(ConsentDetail consentDetail) {
         return signedObject.getSignature();
     }
 
-    private ConsentArtefact mapToConsentArtefact(ConsentRequestDetail consentRequest, GrantedConsent consent) {
+    private ConsentArtefact mapToConsentArtefact(ConsentRequestDetail requestDetail, GrantedConsent granted) {
+        PatientLinkedContext linkedPatientInfo = PatientLinkedContext.builder()
+                .id(requestDetail.getPatient().getId())
+                .careContexts(getLinkedCareContext(granted, requestDetail.getPatient().getId()))
+                .build();
         String consentArtefactId = UUID.randomUUID().toString();
+        //TODO: need to store also the CC
         return ConsentArtefact.builder()
-                .id(consentArtefactId)
-                .requestId(consentRequest.getRequestId())
-                .createdAt(consentRequest.getCreatedAt())
-                .purpose(consentRequest.getPurpose())
-                .patient(consentRequest.getPatient())
-                .hiu(consentRequest.getHiu())
-                .requester(consentRequest.getRequester())
-                .hip(consent.getHip())
-                .hiTypes(consent.getHiTypes())
-                .permission(consent.getPermission())
+                .consentId(consentArtefactId)
+                .createdAt(new Date())
+                .purpose(requestDetail.getPurpose())
+                .patient(linkedPatientInfo)
+                .hiu(requestDetail.getHiu())
+                .requester(requestDetail.getRequester())
+                .hip(granted.getHip())
+                .hiTypes(granted.getHiTypes())
+                .permission(granted.getPermission())
                 .build();
     }
 
+    private LinkedContext[] getLinkedCareContext(GrantedConsent granted, String id) {
+        //TODO - should include HIP patient reference
+        LinkedContext[] linkedContexts = granted.getCareContexts().stream().map(cc -> {
+            return LinkedContext.builder().contextReference(cc.getContextReference()).build();
+        }).toArray(LinkedContext[]::new);
+        return linkedContexts;
+    }
+
     private Mono<ConsentRequestDetail> validateConsentRequest(String requestId) {
         return consentRequestRepository.requestOf(requestId, ConsentStatus.REQUESTED.toString())
                 .switchIfEmpty(Mono.error(ClientError.consentRequestNotFound()));
     }
+
+    public Mono<ConsentArtefactRepresentation> getConsent(String consentId, String hiuId) {
+        return consentArtefactRepository.getConsentArtefact(consentId).doOnSuccess(r -> {
+            if (r == null) {
+                Mono.error(ClientError.consentArtefactNotFound());
+            } else if (!r.getConsentDetail().getHiu().getId().equals(hiuId)) {
```

Comment:
Could we fetch the consent artefact from repository by consentId + hiuId only if possible?

Reply:
We are not storing the hiuid as a column. potentially we could query the JSON in future, or introduce a specific column for the hiu_id